classes/submission/PKPSubmission.inc.php

```diff
@@ -219,6 +219,20 @@ abstract public function getSectionId();
      */
     abstract public function _getContextLicenseFieldValue($locale, $field);
 
+    /**
+     * Whether the submission has a DOI associated with it in any way (e.g. publication or other)
+     *
+     */
+    public function hasDoi(): bool
+    {
+        $retVal = false;
+        $publication = $this->getCurrentPublication();
+        if ($publication->getData('doi') !== null) {
+            return true;
+        }
+        return $retVal;
+    }
+
```

Comment:
This looks like code that can be traced to the pub objects export grid, so can probably be removed.